core/src/components/chip/chip.scss

```diff
@@ -67,33 +65,32 @@
 }
 
 :host(.chip-outline) {
-  border-color: rgba($text-color-rgb, .32);
+  border-color: rgba($text-color-rgb, 0.32);
 
   background: transparent;
 }
 
 :host(.chip-outline.ion-color) {
-  border-color: current-color(base, .32);
+  border-color: current-color(base, 0.32);
 }
 
 :host(.chip-outline:not(.ion-color):focus) {
-  background: rgba($text-color-rgb, .04);
+  background: rgba($text-color-rgb, 0.04);
 }
 
 :host(.chip-outline.ion-activated:not(.ion-color)) {
-  background: rgba($text-color-rgb, .08);
+  background: rgba($text-color-rgb, 0.08);
 }
 
-
 // Chip Icon
 // ---------------------------------------------
 
 ::slotted(ion-icon) {
-  font-size: 20px;
+  font-size: 1.25em;
```

Comment:
This isn't quite the same as what 20px was doing. `1.25em` is going to use the computed font-size of `ion-chip` as the basis. Given a root font size of `16px`, the `ion-chip` in your PR will have a computed font size of `14px`. 

1.25 * 14 = 17.5px which means `ion-icon` will be smaller than in `main`. This is likely why you are seeing changed screenshots. (Same goes for the proposed `ion-icon` change).

In this case, you probably want to stick with `rem` instead of `em`.

Reply:
Can't we use `clamp` to enforce a min/max?

i..e `clamp(13px, 1.25rem, 22px)` I made the max/min values up, but that's the general idea.

Reply:
yes, we can do that. but if we do, we have to separate back out the ios & md styles for these. (because the md ones don't have a max or min.) using em seemed simpler.

Reply:
hmm so maybe I'm not fully understanding what you are trying to accomplish with getting the icons/avatars to scale.

My understanding of the design document is both the text and the icons should scale according to the root font size. On iOS, both the text and the icons should have max/min font sizes.

The reason why I originally commented was the proposed change is causing visual regressions which is going to cause layout changes in developer applications.

Could you clarify what you are trying to accomplish? I think I'll be able to give helpful feedback once I understand this a bit better.

Reply:
option 1: how it is in this PR right now. 

note: this should not change anything visually from how it was, at regular font sizes. in my browser, the icons and avatars are precisely the same size they were before this PR. i'm not sure what's up with the screenshots. i don't visually see a difference in the screenshots either.

```scss
// ios only
:host {
  font-size: clamp(13px, 0.875rem, 22px);
}

// md only
:host {
  font-size: 0.875rem;
}

// ios & md
::slotted(ion-icon) {
  font-size: calc(20em / 14);
}
::slotted(ion-avatar) {
  width: calc(24em / 14);
  height: calc(24em / 14);
}
```

option 2

```scss
// ios only
:host {
  font-size: clamp(13px, 0.875rem, 22px);
}
::slotted(ion-icon) {
  // we will have to remember to change the 19 and/or 31 here if we change 13 and/or 22 above... and remember how to calculate them
  // alternatively, we can make 13px and 22px variables above and utilize those here, but that makes it harder to read
  font-size: clamp(19px, 1.25rem, 31px);
}
::slotted(ion-avatar) {
  // we will have to remember to change the 22 and/or 38 here if we change 13 and/or 22 above... and remember how to calculate them
  width: clamp(22px, 1.5rem, 38px);
  height: clamp(22px, 1.5rem, 38px);
}

// md only
:host {
  font-size: 0.875rem;
}
::slotted(ion-icon) {
  font-size: 1.25rem;
}
::slotted(ion-avatar) {
  width: 1.5rem;
  height: 1.5rem;
}
```

i prefer option 1, because of the reasons in the comments of option 2 and because option 1 is simpler and more similar between ios & md.

Reply:
Ok I think I understand now. You want to size the icons relative to the text initially to avoid visual diffs, but you also want it to scale according to the parent font size (i.e. the size of the text) and account for the max/min font sizes.

I think the confusing part for me is it's not immediately clear where the "14" comes from since the text font size is a rem not a pixel value anymore.

I think we could use some sass variables to make this a bit easier to understand.

Some rough code, but this might work:

```scss
$chip-font-size: 14;
$avatar-font-size: math.div(24em, $chip-font-size);

::slotted(ion-avatar) {
  flex-shrink: 0;

  width: $avatar-font-size;
  height: $avatar-font-size;
}
```

This way it's clear that the 14 is referring to the original font size. The other benefit is we can do these calculations on build and not at runtime (which is what `calc` was doing). I can think on this more tomorrow to see if there are other ways to do this.

Reply:
As someone who was also confused on the "14", I really like the idea of variables that Liam is proposing. So +1 on that.

Reply:
@liamdebeasi @thetaPC this does make sense to me. i updated the PR